ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -233,6 +281,25 @@ let all (lookup: string -> string option) (list: string -> string list) ~__conte
 	    Db.VM_guest_metrics.set_other ~__context ~self:gm ~value:other;
 	    Helpers.call_api_functions ~__context (fun rpc session_id -> Client.Client.VM.update_allowed_operations rpc session_id self);
 	  end;
+	  if(vif_state_cached <> vif_state) then begin
```

Comment:
Thanks Thomas.

To get `network_paths_optimzied`, the following steps must be run:
1.  get `vif_state` which contains the state of all VIFs;
2.  execute logical AND of the states.
That means under each `vm_guest_metrics` refresh, to compare `network_paths_optimized` and `network_paths_optimized_cached`,
`network_paths_optimized` must be calculated with both step 1 and 2 even if `vif_state` has no change.

If we just cache `vif_state`, step 1 is mandatory with `vm_guest_metrics` refresh.
But only when `vif_state` doesn't equal to `vif_state_cached`, step 2 will be executed.

Considering `vif_state` is normally immutable, step 2 seldom needs to run.
For VBD, that's similar.

Could you please let me know if you have any concern?
Thanks.